extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -93,6 +109,12 @@ private CommonPanacheQueryImpl(CommonPanacheQueryImpl<?> previousQuery, String n
         return new CommonPanacheQueryImpl<>(this, select.toString() + query, "select count(*) " + query);
     }
 
+    public void filter(String filterName, Map<String, Object> parameters) {
```

Comment:
Maybe use `withFilter` as we already have `withHint` and `withLock` ?

Reply:
We could argue about this. `hint` doesn't represent an action: we don't want the result to be a hint, and `lock` has an immediacy that isn't true, because the results are locked later.
OTOH, `filter` is an action that doesn't sound wrong even if applied immediately.

So I could argue for `filter` instead of `withFilter` but the reasoning is not obvious, and `withFilter` would be more regular. OTOH, we have `page` and `range` and not `atPage` and `atRange`.

So, not sure at all either way, TBH.

Reply:
OK, no need to argue, name it as you prefere ;)
Naming is hard ...

Reply:
I'll leave it like this, but I don't feel strongly either way.